types/addn_names.pp

```diff
@@ -0,0 +1,9 @@
+# a type to describe addn_names
+type K8s::Addn_names = Array[
```

Comment:
I'd personally prefer this to be something like `K8s::SAN`/`K8s::Tls_altnames` or something alike, since them being "Additional names" is usage and not definition